src/util/README.md

```diff
@@ -329,3 +329,18 @@ serialisation queries save only that integer hash code of the string.
 
 \subsubsection irep-serialization-ireps Serialization of `irept` Instances
 
+
+\subsection CProver-output CProver output - printing
+Cprover has 3 principal types of output:
+
+* direct output via std::cout & std::cerr (these are in the process of being removed).
+
+* output from invariants / exceptions
+Invariants output to std::cerr - and provide a backtrace and optionally some diagnostic information.
+For more information on invariants, see `invariant.h`
+Exceptions also print to std::cerr - and have a standard 'what()' interface. More information on exceptions can be found in `exception_utils.h`.
+
+* messages - filtered by a 'verbosity' level.
```

Comment:
@tautschnig 

int parse_options_baset::main()
catch(const invalid_command_line_argument_exceptiont &e)
  {
    std::cerr << e.what() << "\n";
    return CPROVER_EXIT_USAGE_ERROR;
  }
  catch(const cprover_exception_baset &e)
  {
    std::cerr << e.what() << '\n';
    return CPROVER_EXIT_EXCEPTION;
  }

Otherwise it seems that we are going through a message handler (verified via a quick search - not foolproof 'std::cerr ?<<.+what' ).